### PR TITLE
(#83) use shared Paths.TempDirPath in all projects

### LINKED ISSUE #83: Get rid of `General.temppath`
Currently, all three of our executables use `General.temppath`.

We should migrate that to the common `Paths` class, and make sure we use a correct modern approach to temp folder creation.

### PR DESCRIPTION
**Pull request by KeterSCP**
Closes #83

### CODE REVIEW

File: Source/Shared/Paths.cs [hunk 1]
```diff
@@ -82,4 +82,7 @@ private static string GetAppBaseDir()
 
     /// <summary>Directory for screenshots. Write access.</summary>
     public static readonly string ScreenshotsDirPath = Path.Combine(AllPurposeDirPath, "Screenshots");
+
+    /// <summary>Directory for temporary data. Read + write access.</summary>
+    public static readonly string TempDirPath = Directory.CreateTempSubdirectory().FullName;
```
Comment (ForNeVeR) [inline Source/Shared/Paths.cs:87]: Let's pass a `prefix: "Bloodmasters"`, shall we?

File: Source/Tests/Paths/TempDirPathTests.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+﻿namespace Bloodmasters.Tests.Paths;
```
Comment (ForNeVeR) [inline Source/Tests/Paths/TempDirPathTests.cs:1]: I bet this file includes BOM. Let's figure out what creates it in your environment :)

What IDE/editor do you use? And how are you creating the files?

Comment (KeterSCP) [reply Source/Tests/Paths/TempDirPathTests.cs:1]: I'm using Rider. And i found corresponding setting:

image

I didn't touch it, that is a default value

Comment (KeterSCP) [reply Source/Tests/Paths/TempDirPathTests.cs:1]: Changed it to always create without BOM

Comment (ForNeVeR) [reply Source/Tests/Paths/TempDirPathTests.cs:1]: Ok, set it up to no BOM. And this is stupid. Vote for [IDEA-257119](https://youtrack.jetbrains.com/issue/IDEA-257119).

File: Source/Shared/ArchiveManager.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public class ArchiveManager
 		private static Dictionary<string, Archive> archives = new();
 
 		// Temporary path
-		private static string temppath;
+		private static string tempPath = Paths.TempDirPath;
```
Comment (KeterSCP) [inline Source/Shared/ArchiveManager.cs:18]: Thought it would be more readable having it as field here

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**